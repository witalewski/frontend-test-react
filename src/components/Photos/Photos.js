import React from 'react';

const Photos = ({ photos }) => (
  <ul className="photos">
    {photos.map(({ url, src }) => {
      const image = <img alt="Restaurant" src={src} />;
      return (
        <li key={src}>
          {url ? (
            <a href={url} target="_blank" rel="noopener noreferrer">
              {image}
            </a>
          ) : (
            image
          )}
        </li>
      );
    })}
  </ul>
);

export { Photos };
