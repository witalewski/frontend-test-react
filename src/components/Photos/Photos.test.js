import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MOCK_RESTAURANT from '../../fixtures/restaurant.json';
import { Photos } from './Photos';

describe('Photos', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Photos photos={MOCK_RESTAURANT.photos.map(src => ({src}))} />
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
