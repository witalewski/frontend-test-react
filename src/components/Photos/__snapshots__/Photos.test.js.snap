// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos matches snapshot 1`] = `
<ul
  className="photos"
>
  <li>
    <img
      alt="Restaurant"
      src="https://s3-media2.fl.yelpcdn.com/bphoto/zmRMxckIj4rhutGrSRcvYA/o.jpg"
    />
  </li>
  <li>
    <img
      alt="Restaurant"
      src="https://s3-media1.fl.yelpcdn.com/bphoto/T5WhG8Q7cjXNeOu8OteB1Q/o.jpg"
    />
  </li>
  <li>
    <img
      alt="Restaurant"
      src="https://s3-media1.fl.yelpcdn.com/bphoto/vqOmlb109NiG1x54E4UAWg/o.jpg"
    />
  </li>
</ul>
`;
